src/MultiThreadedDemo.java

```diff
@@ -1,24 +1,39 @@
+import sun.security.util.BigInt;
+
+import java.math.BigInteger;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 
 public class MultiThreadedDemo {
 	//Initialize constants
 	public static final int MAXTHREADS = 7;
 	public static final long TOTALPRIMENUMS = 10000;
+	public static final BigInteger MAXNUMTOCHECK = BigInteger.valueOf(Long.MAX_VALUE + 1);
 	public static final int SLEEPTIME = 500;
+	public static final boolean USEBIGINT = false;
 	
 	public static void main(String[] args) {
 		//Initialize Executor Service and make it handle MAXTHREADS at a time.
 		ExecutorService pool = Executors.newFixedThreadPool(MAXTHREADS);
-		
-		//Start a thread for each number you want to check if it's prime
-		for(long i = 1; i <= TOTALPRIMENUMS; i++){
-			//Initialize PrimeWorker with value of i
-			PrimeWorker worker = new PrimeWorker(i);
-			//Give the worker to the pool to execute
-			pool.execute(worker);
+
+		if(USEBIGINT){
```

Comment:
I left is as false so that the program runs using the `long` variable. If you want to run the program using the `BigInteger` variable you just need to edit the final before running the program.